src/game/Entities/TemporarySpawn.cpp

```diff
@@ -174,7 +174,7 @@ void TemporarySpawn::Update(const uint32 diff)
             }
             if (!GetCharmerGuid())
             {
-                if (IsExpired())
+                if (IsExpired() && !IsInCombat())
```

Comment:
Can confirm this behavior with ai findings related to npc Dread Tactician 16959 using s.33679 (15000 duration), he can have multiple adds up during combat which will be despawned on combat end/reached home.

![grafik](https://user-images.githubusercontent.com/19734826/219691381-079df1a3-5cdf-4bf2-a24f-118c364a2af8.png)